### ISSUE TITLE
Empty file & gulp-sourcemaps never completes

### ISSUE THREAD

**Issue by TuckerWhitehouse** [plus3network/gulp-less#122]
If I try to compile an empty less file and use gulp-sourcemaps, the task starts but never completes.

**Comment by stevelacy** [comment]
Please add the following:

- gulpfile
- gulp, gulp-less, and gulp-sourcemap versions

**Comment by TuckerWhitehouse** [comment]
```json
"gulp": "^3.8.10",
"gulp-less": "^2.0.1",
"gulp-load-plugins": "^0.8.0",
"gulp-sourcemaps": "^1.3.0"
```

```js
var gulp = require('gulp');
var plugins = require('gulp-load-plugins')();

gulp.task('less', function () {
  gulp.src('public/**/*.less', { base: '.' })
    .pipe(plugins.sourcemaps.init())
    .pipe(plugins.less())
    .pipe(plugins.sourcemaps.write())
    .pipe(gulp.dest('.'));
});
```

Also tried without ```gulp-load-plugins```, but it made no difference.
```js
var gulp = require('gulp');
var less = require('gulp-less');
var sourcemaps = require('gulp-sourcemaps');

gulp.task('less', function () {
  gulp.src('public/**/*.less', { base: '.' })
    .pipe(sourcemaps.init())
    .pipe(less())
    .pipe(sourcemaps.write())
    .pipe(gulp.dest('.'));
});
```

**Comment by stevelacy** [comment]
Can you add a example `.less` file?

**Comment by TuckerWhitehouse** [comment]
The issue presents itself when compiling an empty less file.

**Comment by TuckerWhitehouse** [comment]
I created a simple gist that I hope helps demonstrate the issue. I added an error listener and include the output from that in the gist.
https://gist.github.com/TuckerWhitehouse/9181bf9bb31b0fb40617

**Comment by HaNdTriX** [comment]
I am having the same issue. 
Does any of you know a proper workaround?

**Comment by HaNdTriX** [comment]
For now I resoved the issue on empty files with [gulp-clip-empty-files](https://www.npmjs.com/package/gulp-clip-empty-files).

    gulp.task('styles:dev', function() {
      return gulp.src(src)
        .pipe(watch(src))

        // Remove empty files from stream
        // To prevent sourcemaps less issue
        // https://github.com/plus3network/gulp-less/issues/122
        .pipe(clip())

        // Report errors but don't disconnect the stream
        .pipe(plumber())

        // Initialize sourcemaps
        .pipe(sourcemaps.init())

        // Compile less
        .pipe(less({
          plugins: [autoprefix]
        }))

        // Write out the sourcemaps
        .pipe(sourcemaps.write())

        // Finally, output the files to the build directory
        .pipe(gulp.dest(dest))

        // Notify Livereload
        .pipe(livereload());

    });
    
But this doesn't work if the content of the file is an empty css rule:

    body{
        
    }

**Comment by contra** [comment]
@HaNdTriX Does the autoprefix plugin support sourcemaps? Try using gulp-autoprefixer instead. I had the same issue with gulp-stylus where some of the plugins were mangling sourcemaps

**Comment by HaNdTriX** [comment]
I am using [less-plugin-autoprefix](https://www.npmjs.com/package/less-plugin-autoprefix) and it works fine. 

The issue also exist without less plugins.

**Comment by contra** [comment]
@HaNdTriX Okay, well I have received issues that say sourcemaps in less-plugin-autoprefix are broken. Could you double check?

**Comment by TuckerWhitehouse** [comment]
The original issue I filed was without less-plugin-autoprefix, and the addition of less-plugin-autoprefix or gulp-autoprefix does not appear to have any impact.

**Comment by contra** [comment]
Can you try this in 2.0.3? I think it was an error reporting issue

**Comment by contra** [comment]
This got fixed in https://github.com/less/less.js/pull/2439

**Comment by heikki** [comment]
Looks like it's broken different way now:

```sh
∴ gulp-less-test gulp less
[09:32:06] Using gulpfile ~/Desktop/gulp-less-test/gulpfile.js
[09:32:06] Starting 'less'...
[09:32:06] Finished 'less' after 8.65 ms

undefined:1
undefined
^
SyntaxError: Unexpected token u
    at Object.parse (native)
    at /Users/heikki/Desktop/gulp-less-test/node_modules/gulp-less/node_modules/accord/lib/adapters/less.js:48:32
    at /Users/heikki/Desktop/gulp-less-test/node_modules/gulp-less/node_modules/less/lib/less/render.js:35:17
    at /Users/heikki/Desktop/gulp-less-test/node_modules/gulp-less/node_modules/less/lib/less/parse.js:63:17
    at Object.finish [as _finish] (/Users/heikki/Desktop/gulp-less-test/node_modules/gulp-less/node_modules/less/lib/less/parser/parser.js:183:28)
    at Object.ImportVisitor._onSequencerEmpty (/Users/heikki/Desktop/gulp-less-test/node_modules/gulp-less/node_modules/less/lib/less/visitors/import-visitor.js:35:14)
    at ImportSequencer.tryRun (/Users/heikki/Desktop/gulp-less-test/node_modules/gulp-less/node_modules/less/lib/less/visitors/import-sequencer.js:50:14)
    at Object.ImportVisitor.run (/Users/heikki/Desktop/gulp-less-test/node_modules/gulp-less/node_modules/less/lib/less/visitors/import-visitor.js:29:25)
    at Object.Parser.parse (/Users/heikki/Desktop/gulp-less-test/node_modules/gulp-less/node_modules/less/lib/less/parser/parser.js:189:22)
    at Object.parse (/Users/heikki/Desktop/gulp-less-test/node_modules/gulp-less/node_modules/less/lib/less/parse.js:61:18)
```

```json
{
  "dependencies": {
    "gulp": "^3.8.11",
    "gulp-less": "^3.0.0",
    "gulp-sourcemaps": "^1.3.0"
  }
}
```

```js
var gulp = require('gulp');
var less = require('gulp-less');
var sourcemaps = require('gulp-sourcemaps');

gulp.task('less', function () {
  gulp.src('public/**/*.less', { base: '.' })
    .pipe(sourcemaps.init())
    .pipe(less())
    .pipe(sourcemaps.write())
    .pipe(gulp.dest('.'));
});
```

**Comment by contra** [comment]
@jenius looks like an accord bug to me

**Comment by contra** [comment]
@jenius https://github.com/jenius/accord/blob/master/lib/adapters/less.coffee#L24

**Comment by heikki** [comment]
@contra `if (options.sourceMap && res.map) {` on line above seems to make it pass.

**Comment by contra** [comment]
@heikki Maybe send a PR? Would get it fixed a lot quicker

**Comment by jenius** [comment]
Thanks for the PR, fixed and shipped as `accord@0.15.1`
